edc-extensions/hashicorp-vault/src/main/java/org/eclipse/dataspaceconnector/core/security/hashicorp/HashicorpVaultExtension.java

```diff
@@ -0,0 +1,121 @@
+/*
+ *  Copyright (c) 2022 Mercedes-Benz Tech Innovation GmbH
+ *
+ *  This program and the accompanying materials are made available under the
+ *  terms of the Apache License, Version 2.0 which is available at
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       Mercedes-Benz Tech Innovation GmbH - Initial API and Implementation
+ *
+ */
+
+package org.eclipse.dataspaceconnector.core.security.hashicorp;
+
+import java.time.Duration;
+import okhttp3.OkHttpClient;
+import org.eclipse.dataspaceconnector.spi.EdcException;
+import org.eclipse.dataspaceconnector.spi.EdcSetting;
+import org.eclipse.dataspaceconnector.spi.monitor.Monitor;
+import org.eclipse.dataspaceconnector.spi.security.CertificateResolver;
+import org.eclipse.dataspaceconnector.spi.security.PrivateKeyResolver;
+import org.eclipse.dataspaceconnector.spi.security.Vault;
+import org.eclipse.dataspaceconnector.spi.security.VaultPrivateKeyResolver;
+import org.eclipse.dataspaceconnector.spi.system.ServiceExtensionContext;
+import org.eclipse.dataspaceconnector.spi.system.VaultExtension;
+
+public class HashicorpVaultExtension implements VaultExtension {
+
+  @EdcSetting(required = true)
+  public static final String VAULT_URL = "edc.vault.url";
+
+  @EdcSetting public static final String VAULT_TOKEN = "edc.vault.token";
```

Comment:
Isn't this also required?
```suggestion
  @EdcSetting(required = true)
  public static final String VAULT_TOKEN = "edc.vault.token";

```